src/main/java/games/strategy/triplea/delegate/Matches.java

```diff
@@ -665,36 +665,47 @@ private Matches() {}
     return Match.of(obj -> unitTypeIsAaForAnything().match(obj.getType()));
   }
 
-  public static final Match<Unit> UnitIsNotAA = unitIsAaForAnything().invert();
+  public static Match<Unit> unitIsNotAa() {
+    return unitIsAaForAnything().invert();
+  }
 
-  public static final Match<UnitType> UnitTypeMaxAAattacksIsInfinite =
-      Match.of(obj -> UnitAttachment.get(obj).getMaxAAattacks() == -1);
+  private static Match<UnitType> unitTypeMaxAaAttacksIsInfinite() {
+    return Match.of(obj -> UnitAttachment.get(obj).getMaxAAattacks() == -1);
+  }
 
-  public static final Match<Unit> UnitMaxAAattacksIsInfinite =
-      Match.of(obj -> UnitTypeMaxAAattacksIsInfinite.match(obj.getType()));
+  static Match<Unit> unitMaxAaAttacksIsInfinite() {
+    return Match.of(obj -> unitTypeMaxAaAttacksIsInfinite().match(obj.getType()));
+  }
 
-  public static final Match<UnitType> UnitTypeMayOverStackAA =
-      Match.of(obj -> UnitAttachment.get(obj).getMayOverStackAA());
+  private static Match<UnitType> unitTypeMayOverStackAa() {
+    return Match.of(obj -> UnitAttachment.get(obj).getMayOverStackAA());
+  }
 
-  public static final Match<Unit> UnitMayOverStackAA = Match.of(obj -> UnitTypeMayOverStackAA.match(obj.getType()));
+  static Match<Unit> unitMayOverStackAa() {
+    return Match.of(obj -> unitTypeMayOverStackAa().match(obj.getType()));
+  }
 
-  public static final Match<Unit> UnitAttackAAisGreaterThanZeroAndMaxAAattacksIsNotZero = Match.of(obj -> {
-    final UnitAttachment ua = UnitAttachment.get(obj.getType());
-    return ua.getAttackAA(obj.getOwner()) > 0 && ua.getMaxAAattacks() != 0;
-  });
+  static Match<Unit> unitAttackAaIsGreaterThanZeroAndMaxAaAttacksIsNotZero() {
+    return Match.of(obj -> {
+      final UnitAttachment ua = UnitAttachment.get(obj.getType());
+      return ua.getAttackAA(obj.getOwner()) > 0 && ua.getMaxAAattacks() != 0;
+    });
+  }
 
-  public static final Match<Unit> UnitOffensiveAttackAAisGreaterThanZeroAndMaxAAattacksIsNotZero = Match.of(obj -> {
-    final UnitAttachment ua = UnitAttachment.get(obj.getType());
-    return ua.getOffensiveAttackAA(obj.getOwner()) > 0 && ua.getMaxAAattacks() != 0;
-  });
+  static Match<Unit> unitOffensiveAttackAaIsGreaterThanZeroAndMaxAaAttacksIsNotZero() {
+    return Match.of(obj -> {
+      final UnitAttachment ua = UnitAttachment.get(obj.getType());
+      return ua.getOffensiveAttackAA(obj.getOwner()) > 0 && ua.getMaxAAattacks() != 0;
+    });
+  }
 
-  public static final Match<Unit> UnitIsInfantry = Match.of(obj -> {
-    final UnitType type = obj.getUnitType();
-    final UnitAttachment ua = UnitAttachment.get(type);
-    return ua.getIsInfantry();
-  });
+  public static Match<Unit> unitIsInfantry() {
+    return Match.of(obj -> UnitAttachment.get(obj.getUnitType()).getIsInfantry());
+  }
```

Comment:
There was some manual refactoring done here in addition to the field -> method conversion.